### DISCUSSION TITLE
Applying CSS to scoped component of Quasar element

### DISCUSSION THREAD

**Discussion by dost-mmg** [quasarframework/quasar#12683] [General - Components / Directives / etc]
I am working on a quasar/vue app. I want to style the dialog popup within one component. I'm using scoped CSS, and if the CSS is not scoped, the style works. If the CSS is scoped, the CSS does not work. I only want to style this dialog in this one component.

The CSS element is:

<style scoped>
 .q-dialog__inner {
    width: 400px;
    background-color: red;
  }

</style>
This does not work:

:deep(.q-dialog__inner) {
  width: 400px !important;
  background-color: red;
 }
I noticed that the global quasar style is marked with !important

**Comment by MilosPaunovic** [comment]
Add an `id` or a `class` to that particular dialog and target it specifically through CSS.

**Reply by dost-mmg** [reply]
How do I assign an Id to the particular dialog where I'm calling it though q-select?

**Reply by MilosPaunovic** [reply]
Nowhere before was mentioned you're caliing it throught select. Create a Codepen repoduction of your issue so someone could tackle it.

**Reply by dost-mmg** [reply]
Yes, right -- sorry about that. Here is a codepen: https://codepen.io/kiggs1881/pen/oNoOzEj

**Comment by MilosPaunovic** [comment]
As I can see, there is no easy way to acomplish this.

For now, you could use `.q-select__dialog` class without scoped attribute to narrow it a little, but perhaps some prop for `QSelect` like `dialog-content-class` could be introduced.
